src/main/entities/NewPassword.java

```diff
@@ -0,0 +1,146 @@
+            case 3 -> choice = PasswordGenerator.generateSymbols(1);
+            case 4 -> choice = PasswordGenerator.generateNumbers(1);
+            default -> {
+            }
+        }
+        password += choice;
+        return password;
+    }
+
+    // When only Two Choices are selected
+    public static String twoChoices(String code, int len){
+        String password = "";
+        for(int i=0;i<len;i++){
+            double binar = (Math.random()*(2))+0;     // Random number 1 or 2
+            double one = (Math.random()*(2-1))+1;       // Random number 1
+            double two = (Math.random()*(3-2))+2;       // Random number 2
+            double three = (Math.random()*(4-3))+3;     // Random number 3
+            double four = (Math.random()*(5-4))+4;      // Random number 4
+            double finalans = switch (code) {
+                case "LS", "SL" -> (int) binar == 0 ? one : three;
+                // When lowercase letters checkbox & numbers checkbox are selected
+                case "LN", "NL" -> (int) binar == 0 ? one : four;
+                // When lowercase letters checkbox & uppercase letters' checkbox are selected
+                case "LU", "UL" -> (int) binar == 0 ? one : two;
+                // When uppercase letters checkbox & numbers checkbox are selected
+                case "UN", "NU" -> (int) binar == 0 ? two : four;
+                // When symbols checkbox & uppercase letters' checkbox are selected
+                case "SU", "US" -> (int) binar == 0 ? three : two;
+                // When numbers checkbox & symbols checkbox are selected
+                case "NS", "SN" -> (int) binar == 0 ? four : three;
+                default -> 0;
+                //When lowercase letters checkbox & symbols checkbox are selected
+            };
+            int num = (int)finalans;
+            String choice = "";
+            password = getString(password, num, choice);
+        }
+        return password;
+    }
+
+    // When only Three Choices are selected
+    public static String threeChoices(String code, int len){
+        String password = "";
+        for(int i=0;i<len;i++){
+            double trio = (Math.random()*(3))+0;
+            double one = (Math.random()*(2-1))+1;
+            double two = (Math.random()*(3-2))+2;
+            double three = (Math.random()*(4-3))+3;
+            double four = (Math.random()*(5-4))+4;
+            double finalans=0;
+            switch (code) {
+                case "LUN":
+                case "ULN":
+                case "NLU":
+                    if ((int) trio % 2 == 0) {
+                        finalans = (int) trio == 0 ? one : two;
+                    } else {
+                        finalans = (int) trio == 0 ? one : four;
+                    }
+                    break;
+                case "LUS":
+                case "USL":
+                case "SLU":
+                    if ((int) trio % 2 == 0) {
```

Comment:
These if statements all look the same except the result values. Can they be extracted to a helper method?

Reply:
I need your insight on this one. The only way I see this getting replaced is to transfer the entire code-chunk to a helper function and give it the conditions string.

Reply:
If you select the if statement in IntelliJ and do Refactor->Extract->Method that will give you the basic structure of the helper, but you'll need to generalize it (i.e. take in the four possible results of the nested conditionals rather than the three, since the pattern differs in the last case).